xbmc/Application.cpp

```diff
@@ -1978,7 +1979,34 @@ bool CApplication::OnAction(const CAction &action)
     return true;
   }
 
-  if ((action.GetID() == ACTION_INCREASE_RATING || action.GetID() == ACTION_DECREASE_RATING) && m_appPlayer.IsPlayingAudio())
+  if ((action.GetID() == ACTION_SET_RATING) && m_appPlayer.IsPlayingAudio())
+  {
+    const CMusicInfoTag *tag = g_infoManager.GetCurrentSongTag();
```

Comment:
Capp should be the leadng source. why do you pull info from g_infoManager (gui)? the won't be a GUIInfoManager if running headless.

Reply:
Responding to the current, all be it non ideal,  situation. If we have a GUI, and that is still the majority use, the GUI item tag values can end up different from that of the app item.  
At least with this the OSD rating selection dialog shows the same as the rest of the OSD.

Reply:
Also existing code for `ACTION_INCREASE_RATING` already does same

Reply:
this does not change the fact that this is wrong. instead of continuing doing it wrong, it should be corrected. Capp is leading source and must not pull info from guiInfo